src/ethereum_test_specs/helpers.py

```diff
@@ -0,0 +1,100 @@
+from ethereum_test_types import Transaction
+
+
+@dataclass
+class TransactionExceptionInfo:
+    """Info to print transaction exception error messages"""
+
+    t8n_error_message: str | None
+    transaction_ind: int
+    tx: Transaction
+
+
+def verify_transaction_exception(
+    exception_mapper: ExceptionMapper, info: TransactionExceptionInfo
+):
+    """Verify transaction exception"""
+    exception_info = f"TransactionException (pos={info.transaction_ind}, nonce={info.tx.nonce})\n"
+    expected_error: bool = info.tx.error is not None or (
+        isinstance(info.tx.error, list) and len(info.tx.error) != 0
+    )
+
+    # info.tx.error is expected error code defined in .py test
+    if expected_error and not info.t8n_error_message:
+        raise Exception(f"{exception_info} Error: tx expected to fail succeeded")
+    elif not expected_error and info.t8n_error_message:
+        raise Exception(
+            f'{exception_info} Error: tx unexpectedly failed: "{info.t8n_error_message}"'
+        )
+    elif expected_error and info.t8n_error_message:
+
+        if isinstance(info.tx.error, List):
+            for expected_exception in info.tx.error:
+                expected_error_msg = exception_mapper.exception_to_message(expected_exception)
+                if expected_error_msg is None:
+                    continue
+                if expected_error_msg in info.t8n_error_message:
+                    # One of expected exceptions is found in tx error string, no error
+                    return
+
+        if isinstance(info.tx.error, List):
+            expected_exception = info.tx.error[0]
+        elif info.tx.error is None:
+            return  # will never happen but removes python logic check
+        else:
+            expected_exception = info.tx.error
+
+        expected_error_msg = exception_mapper.exception_to_message(expected_exception)
+        t8n_error_exception = exception_mapper.message_to_exception(info.t8n_error_message)
+        exception_mapper_name = exception_mapper.__class__.__name__
+
+        define_message_hint = (
+            f"No message defined for {expected_exception}, "
+            f"please add it to {exception_mapper_name}"
+            if expected_error_msg is None
+            else ""
+        )
+        define_exception_hint = (
+            "No exception defined for error message got, "
+            f"please add it to {exception_mapper_name}"
+            if t8n_error_exception == UndefinedException.UNDEFINED_EXCEPTION
+            else ""
+        )
+
+        if expected_error_msg is None or expected_error_msg not in info.t8n_error_message:
+            raise Exception(
+                f"{exception_info}"
+                f"Error: exception mismatch:"
+                f"\n want = '{expected_error_msg}' ({expected_exception}),"
+                f"\n got  = '{info.t8n_error_message}' ({t8n_error_exception})"
+                f"\n {define_message_hint}"
+                f"\n {define_exception_hint}"
```

Comment:
I would also add an Exception class for this and use a mutli-line f-string for the message and apply `textwrap.dedent()` to it. There's examples of `dedent()` usage within EEST.

Reply:
looks like it's ok without dedent. but should keep it in mind